src/transformers/models/dab_detr/configuration_dab_detr.py

```diff
@@ -0,0 +1,338 @@
+        temperature_width (`int`, *optional*, defaults to 20):
+            Temperature parameter to tune the flatness of positional attention (WIDTH)
+        iter_update (`bool`, *optional*, defaults to `True`):
+            Whether to use dynamic iterative anchor updates.
+        query_dim (`int`, *optional*, defaults to 4):
+            Query dimension parameter represents the size of the output vector.
+        bbox_embed_diff_each_layer (`bool`, *optional*, defaults to `False`):
+            Whether to perform layer-by-layer bounding box embedding refinement.
+        decoder_bbox_embed_diff_each_layer (`bool`, *optional*, defaults to `False`):
+            Whether to perform layer-by-layer bounding box embedding refinement.
+        random_refpoints_xy (`bool`, *optional*, defaults to `False`):
+            Whether to fix the x and y coordinates of the anchor boxes with random initialization.
+        keep_query_pos (`bool`, *optional*, defaults to `False`):
+            Whether to concatenate the projected positional embedding from the object query into the original query (key) in every decoder layer.
+        query_scale_type (`str`, *optional*, defaults to `"cond_elewise"`):
+            Scale type options:
+                # 'cond_elewise' - Conditional element-wise scaling using content information.
+                # 'cond_scalar' - Conditional scalar scaling using content information.
+                # 'fix_elewise' - Fixed element-wise scaling.
+        num_patterns (`int`, *optional*, defaults to 0):
+            Number of pattern embeddings.
+        normalize_before (`bool`, *optional*, defaults to `False`):
+            Whether we use a normalization layer in the Encoder or not.
+        sine_position_embedding_normalize (`bool`, *optional*, defaults to `True`):
+            Whether the positional embeddings are normalized and scaled by sine_position_embedding_scale value.
+        sine_position_embedding_scale (`float`, *optional*, defaults to 'None'):
+            Scaling factor applied to the normalized positional encodings.
+        initializer_bias_prior_prob (`float`, *optional*):
+            The prior probability used by the bias initializer to initialize biases for `enc_score_head` and `class_embed`.
+            If `None`, `prior_prob` computed as `prior_prob = 1 / (num_labels + 1)` while initializing model weights.
+
+
+    Examples:
+
+    ```python
+    >>> from transformers import DABDETRConfig, DABDETRModel
+
+    >>> # Initializing a DAB-DETR IDEA-Research/dab_detr-base style configuration
+    >>> configuration = DABDETRConfig()
+
+    >>> # Initializing a model (with random weights) from the IDEA-Research/dab_detr-base style configuration
+    >>> model = DABDETRModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+
+    model_type = "dab-detr"
+    keys_to_ignore_at_inference = ["past_key_values"]
+    attribute_map = {
```

Comment:
Yes, but we need to keep it since some config tests fail because this is the setter for the variable. I haven't looked into it yet, but if it's critical, I will.